cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -2649,10 +2767,7 @@ private ApplicationMeta decodeRow(StructuredRow row) {
     ApplicationMeta meta = GSON.fromJson(
         row.getString(StoreDefinition.AppMetadataStore.APPLICATION_DATA_FIELD),
         ApplicationMeta.class);
-    SourceControlMeta sourceControl = GSON.fromJson(
-        row.getString(StoreDefinition.AppMetadataStore.SOURCE_CONTROL_META),
-        SourceControlMeta.class);
-
+    SourceControlMeta sourceControl = null;
```

Comment:
I'm not really clear on the implications of this change. It seems like it would impact the REST APIs for listing applications and fetching an application? If so that would be a breaking change to the API, which we shouldn't do unless there is a good reason.

On an unrelated note, since this is always null, we don't need the variable and can instead use the ApplicationMeta constructor that doesn't take SourceControlMeta as an argument.

Reply:
ok it looks like it's expected that the real sourceControl is expected to be read from the other table and set in the DefaultStore. What's the plan existing application rows though? Will there be some sort of data migration?

Reply:
Whenever we are listing the apps in the list page (by calling /namespaces/<namespace-id>/apps), the expected behaviour is to not pass/show the source control metadata. Instead it will be displayed in the sync pipeline - Namspace pipelines page ((by calling /namespaces/<namespace-id>/sourcecontrol/apps).

Regarding the migration of data, we were initially thinking of adding a one-time service setup which will block the startup service and copy the required data to the new table. After having discussion with you, you suggested to parallelize the process so that it doesn't block the startup service. So, we will be running the service in the background and handle the edge cases of data getting overridden. PR for migration of data will be raised soon.